partition: add doc for global index, also with GLOBAL IndexOption/ColumnOption (#18543)
pingcap/docs: system-variables.md

### tidb_enable_global_index <span class="version-mark">New in v7.6.0</span>

> **Warning:**
>

> > The feature controlled by this variable is an experimental feature. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

Reviewer: ```suggestion
> The feature controlled by this variable is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `OFF`
- Possible values: `OFF`, `ON`

> - This variable controls whether to support creating [global indexes](/partitioned-table.md#global-indexes) for partitioned tables. When this variable is enabled, TiDB allows to creates unique indexes that do not contain all columns in the partition expressions by using the GLOBAL IndexOption/ColumnOption.

Reviewer 2: ```suggestion
- This variable controls whether to support creating [global indexes](/partitioned-table.md#global-indexes) for partitioned tables. When this variable is enabled, TiDB allows to create unique indexes that do not contain all columns in the partitioning expressions by using the `GLOBAL` IndexOption or ColumnOption.
```

> - This variable controls whether to support creating [global indexes](/partitioned-table.md#global-indexes) for partitioned tables. When this variable is enabled, TiDB allows to create unique indexes that do not contain all columns in the partitioning expressions by using the `GLOBAL` IndexOption or ColumnOption.

Reviewer: ```suggestion
- This variable controls whether to support creating [global indexes](/partitioned-table.md#global-indexes) for partitioned tables. When this variable is enabled, TiDB allows you to create unique indexes that **do not contain all columns in the partitioning expressions** by specifying `GLOBAL` in the index definition.
```
